plugin/trino-influxdb/src/main/java/io/trino/plugin/influxdb/NativeInfluxClient.java

```diff
@@ -0,0 +1,282 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.trino.plugin.influxdb;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Sets;
+import com.google.inject.Inject;
+import io.airlift.log.Logger;
+import io.trino.spi.connector.SchemaTableName;
+import okhttp3.ConnectionPool;
+import okhttp3.OkHttpClient;
+import okhttp3.logging.HttpLoggingInterceptor;
+import org.influxdb.InfluxDB;
+import org.influxdb.InfluxDBFactory;
+import org.influxdb.dto.Query;
+import org.influxdb.dto.QueryResult;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.common.collect.Iterables.getOnlyElement;
+import static io.trino.plugin.influxdb.InfluxConstant.ColumnKind.FIELD;
+import static io.trino.plugin.influxdb.InfluxConstant.ColumnKind.TAG;
+import static io.trino.plugin.influxdb.InfluxConstant.ColumnKind.TIME;
+import static io.trino.plugin.influxdb.InfluxConstant.ColumnName;
+import static io.trino.plugin.influxdb.TypeUtils.TIMESTAMP;
+import static io.trino.plugin.influxdb.TypeUtils.toTrinoType;
+import static io.trino.spi.type.VarcharType.VARCHAR;
+import static java.util.Locale.ENGLISH;
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static okhttp3.logging.HttpLoggingInterceptor.Level.NONE;
+import static org.influxdb.dto.QueryResult.Result;
+import static org.influxdb.dto.QueryResult.Series;
+
+public class NativeInfluxClient
+        implements InfluxClient
```

Comment:
Any plan to add other implementations for InfluxClient?

Reply:
Not yet, but in the future that can be the case. In this PR someone already asked about the compatibility with InfluxDB v2. Which this implementation cannot do.